src/extension.ts

```diff
@@ -1,98 +1,175 @@
-import * as vscode from 'vscode';
-
-import { TelemetryEvent } from './TelemetryEvent';
-import { TelemetryEventQueue } from './TelemetryEventQueue';
-import { Reporter } from './Reporter';
-import { Logger } from './Logger';
-
-const OPTIN_REQUESTED_KEY = 'redhat.telemetry.optInRequested';
-
+import * as vscode from "vscode";
+import { SegmentInitializer } from "./segmentInitializer";
+import { Logger } from "./utils/logger";
+import Analytics from "analytics-node";
```

Comment:
Analytics and SegmentInitializer dependencies should be implementation details of the Reporter, not seen from extension.ts

Reply:
Updated the PR with this change,
Now these implementations are available to "telemetryService", Reporter and no more visible in extension.ts